pkg/auth/tokenparser.go

```diff
@@ -0,0 +1,83 @@
+package auth
+
+import (
+	"errors"
+	"log"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+// TokenClaims represents access token claims
+type TokenClaims struct {
+	Name          string         `json:"name"`
+	Username      string         `json:"preferred_username"`
+	GivenName     string         `json:"given_name"`
+	FamilyName    string         `json:"family_name"`
+	Email         string         `json:"email"`
+	EmailVerified bool           `json:"email_verified"`
+	Company       string         `json:"company"`
+	SessionState  string         `json:"session_state"`
+	Approved      bool           `json:"approved"`
+	jwt.StandardClaims
+}
+
+// TokenParser represents a parser for JWT tokens.
+type TokenParser struct {
+	config     *configuration.Registry
+	logger     *log.Logger
+	keyManager *KeyManager
+}
+
+// NewTokenParser creates a new TokenParser.
+func NewTokenParser(logger *log.Logger, config *configuration.Registry, keyManager *KeyManager) (*TokenParser, error) {
+	if logger == nil {
```

Comment:
IMO having the default TokenParser singleton would make more sense than having the default KeyManager singleton.
You don't need to access KeyManager directly. You need to access the token parser which is using the keys from the manager behind the sense. KeyManager is used only by the token manager. It's rather an implementation detail of the parser. KeyManager alone is useless.

Reply:
But TokenParser can freely instantiated, it does not hold state. This is the difference between it and KeyManager. I would put the singleton as deep as possible to lock as little as possible as a singleton..

Reply:
...but I added that in 9fe2cac to get this done. We have <48 hours to complete the story.

Reply:
The token parser holds the state indirectly (it holds KeyManager). It's all about usage IMO.
KeyManager is supposed to be used by the token parser too. So, it's rather a token parser implementation detail. And if we needed to use the parser in many places then having a default instance you can access everywhere (`auth.DefatultTokenParser()`) would help. But since we don't plan to use it only in the auth MW then I think it's not that important.

> but I added that in 9fe2cac to get this done

I actually didn't mean to have two defaults: KeyManager and TokenParser. I was talking about just one default TokenParser and no any default key manager.
But as I said it's not that important. I'm fine with keeping it as it is now or with reverting 9fe2cac.